statsmodels/tsa/statespace/mlemodel.py

```diff
@@ -962,7 +964,12 @@ def cov_params_cs(self):
         )
         self.model.update(self.params)
 
-        return -np.linalg.inv(nobs * evaluated_hessian)
+        neg_cov, singular_values = pinv_extended(nobs * evaluated_hessian)
+
+        if self._rank is None:
+            self._rank = np.linalg.matrix_rank(np.diag(singular_values))
```

Comment:
Good question - I did it this way because this is the way it's done in `linear_model.py`; I think maybe it has to do with letting `matrix_rank` figure out the appropriate cutoff for zero singular values. We could do it ourselves though.